test/unit/alignment/aligned_sequence_test.cpp

```diff
@@ -22,19 +26,17 @@ class aligned_sequence<container_type> : public ::testing::Test
 public:
     // Initializer function is needed for the typed test because the gapped_decorator
     // will be initialized differently than the naive vector<gapped<dna>>.
-    void initialize_typed_test_container(container_type & container, std::string const && target)
+    void initialise_typed_test_container(container_type & container, dna4_vector const & target)
     {
+        container.clear();
         for (auto & val : target)
         {
-            typename container_type::value_type cval{};
-            assign_char(cval, val);
-            container.push_back(cval);
+            container.push_back(typename container_type::value_type{}.assign_char(to_char(val)));
```

Comment:
Don't rely on member functions in generic contexts!

Reply:
but it is in our test suite with only our alphabets...

Reply:
not all our alphabets do :point_up:

Reply:
:expressionless: